### PR TITLE
airbyte-to-flow: support normalization patch files

### PR DESCRIPTION
**Pull request by williamhbaker**
Closes https://github.com/estuary/airbyte/issues/125

This is a follow-up to https://github.com/estuary/airbyte/pull/124. It provides a generalized way for `airbyte-to-flow` to support data normalization functions, and for individual connectors to apply normalizations to specified properties in their produced documents.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,12 +74,14 @@ and copied in Dockerfile. The following files are supported:
        documentation_url. Expects a single key `documentation_url` with a string value
     5. `streams/<stream-name>.patch.json`: to patch a specific stream's document schema
     6. `streams/<stream-name>.pk.json`: to patch a specific stream's primary key, expects an array of strings
+    7. `normalize.json`: to apply data normalization functions to specific fields of documents generated by the connector. Normalizations are provided as a list (JSON array) of objects with keys `pointer` having a value of the pointer to the document field that the normalization will apply to, and `normalization` having a value of the name of the normalization function to apply to the field at that pointer. Normalization function names should provided as `snake_case`; see `airbyte_to_flow/src/interceptors/normalize.rs` for the supported normalization functions.
```
Comment (mdibaiee) [inline README.md:77]: What do you think about making this stream-specific? i.e. move this file to `streams/<stream-name>.normalize.json`

I'm just thinking of cases where the key we are normalizing for one stream has a very generic name, e.g. `name`, or `date`, or `timestamp`, and we only want to normalize one of the streams, not all of them.

Comment (williamhbaker) [reply README.md]: That seems like a really good idea, and will inevitably be necessary. I pushed a change to allow for that.

File: airbyte-to-flow/src/interceptors/airbyte_source_interceptor.rs [hunk 2]
```diff
@@ -366,7 +370,18 @@ impl AirbyteSourceInterceptor {
                         for (i, binding) in c.bindings.iter().enumerate() {
                             let resource: ResourceSpec =
                                 serde_json::from_str(&binding.resource_spec_json)?;
-                            stb.lock().await.insert(resource.stream.clone(), i);
+
+                            let normalizations = std::fs::read_to_string(format!(
+                                "{}/{}{}",
+                                STREAM_PATCH_DIR_NAME,
+                                &resource.stream,
```
Comment (williamhbaker) [inline airbyte-to-flow/src/interceptors/airbyte_source_interceptor.rs:377]: At this point, I'm pretty sure `resource.stream` is directly usable without needing to do `stream_to_recommended_name`. Let me know if I'm wrong about that.

**Review by mdibaiee — APPROVED** [review]
LGTM, thanks Will! There is some leftovers that I suggested below

**Pull request merged**